feature-push-notifications/src/main/java/io/novafoundation/nova/feature_push_notifications/data/data/subscription/RealPushSubscriptionService.kt

```diff
@@ -167,4 +170,19 @@ class RealPushSubscriptionService(
     private fun Map<String, Any>.nullIfEmpty(): Map<String, Any>? {
         return if (isEmpty()) null else this
     }
+
+    private fun <T> Map<ChainId, T>.transfromChainIdsTo16Hex(): Map<String, T> {
+        return mapKeys { (chainId, _) -> chainId.to16Hex() }
+    }
+
+    private fun List<ChainId>.transfromChainIdsTo16Hex(): List<String> {
+        return map { chainId -> chainId.to16Hex() }
+    }
+
+    private fun ChainId.to16Hex(): String {
+        return fromHex()
+            .take(16)
+            .toByteArray()
+            .toHexString(withPrefix = true)
```

Comment:
We can reduce number of operations by doing `removeHexPrefix.().take(32).requireHexPrefix()`. Also iirc similar code but without final prefix is used in `Caip2Identifier.Polkadot` so we can create a common function to reuse it in both places